opentracing/tracecontext.go

```diff
@@ -51,18 +43,18 @@ type TraceContext interface {
 	// `restrictedKey` MUST match the regular expression
 	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. That is, it must
 	// start with a letter or number, and the remaining characters must be
-	// letters, numbers, or hyphens. See CanonicalizeTraceTagKey(). If
-	// `restrictedKey` does not meet these criteria, SetTraceTag() results in
-	// undefined behavior.
+	// letters, numbers, or hyphens. See CanonicalizeTraceAttributeKey(). If
+	// `restrictedKey` does not meet these criteria, SetTraceAttribute()
+	// results in undefined behavior.
 	//
 	// Returns a reference to this TraceContext for chaining, etc.
-	SetTraceTag(restrictedKey, value string) TraceContext
+	SetTraceAttribute(restrictedKey, value string) TraceContext
```

Comment:
Elsewhere I mentioned that I had people asking me about upstream propagation as well, e.g. for controlling an http cookie from a downstream layer. I don't think it would be technically very difficult to support, subject to some conflict resolution policy. Given that possibility, do we actually need to emphasize downstream-ness of these attributes?

It might be more useful to have an additional dimension to the Trace Attributes that clarifies the propagation semantics: downstream all the way, downstream one level only, upstream, bidirectional, etc. Unfortunately, I don't have concrete use cases just yet for different scenarios, will try to acquire some this week.